src/Cowsay.zig

```diff
@@ -11,23 +11,17 @@ const default_cow =
     \\    ||     ||
 ;
 
-var general_purpose_allocator = std.heap.GeneralPurposeAllocator(.{}){};
-
-
-
 /// writer for output. must be an AnyWriter. Use `.any()` to convert to AnyWriter before passing in.
-w: std.io.AnyWriter,
+writer: std.io.AnyWriter,
 /// the eyes. will substitute the first two `o` of the cow
 eyes: [2]u8 = [2]u8{ 'o', 'o' },
 
-/// allocator to use when formatting message
-allocator: std.mem.Allocator = general_purpose_allocator.allocator(),
```

Comment:
As mentioned in previous discussion comments, an allocator stored in the struct typically indicates that the struct itself stores allocated state that must later be freed in a `deinit` function. If this is not the case and all allocations are only function-local, it is preferable for those functions to take in an allocator directly.